### PR TITLE
v6 version of Polyorphic Events sample with note for topology difference

### PR DESCRIPTION
**Pull request by SeanFeldman**
* Sample converted to v6
* .md update to let users know that this workaround is not needed for the new topology

@Particular/documentation please review

Note: I'm using lower core version since transport is not up-to-date with the latest core unstable

### CODE REVIEW

**Comment by SimonCropp** [comment]
How come still bus configuration?

**Comment by SimonCropp** [comment]
Missing prerelease marker file

File: samples/azure/polymorphic-events-asb/Version_6/Publisher/app.config [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        <bindingRedirect oldVersion="0.0.0.0-5.6.4.0" newVersion="5.6.4.0" />
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="Microsoft.Data.OData" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-5.6.4.0" newVersion="5.6.4.0" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
```
Comment (SimonCropp) [inline samples/azure/polymorphic-events-asb/Version_6/Publisher/app.config]: Remove this

File: samples/azure/polymorphic-events-asb/Version_6/Publisher/app.config [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  </startup>
+  <appSettings>
```
Comment (SimonCropp) [inline samples/azure/polymorphic-events-asb/Version_6/Publisher/app.config]: Remove this

File: samples/azure/polymorphic-events-asb/Version_6/Shared/ConfigErrorQueue.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using NServiceBus.Config;
+using NServiceBus.Config.ConfigurationSource;
+
+class ConfigErrorQueue : IProvideConfiguration<MessageForwardingInCaseOfFaultConfig>
+{
+    public MessageForwardingInCaseOfFaultConfig GetConfiguration()
```
Comment (SimonCropp) [inline samples/azure/polymorphic-events-asb/Version_6/Shared/ConfigErrorQueue.cs]: Move over to code config

File: samples/azure/polymorphic-events-asb/Version_6/Subscriber/app.config [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      <dependentAssembly>
+        <assemblyIdentity name="Microsoft.Data.OData" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-5.6.4.0" newVersion="5.6.4.0" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+  <appSettings>
+  </appSettings>
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
```
Comment (SimonCropp) [inline samples/azure/polymorphic-events-asb/Version_6/Subscriber/app.config]: Remove this

**Comment by SeanFeldman** [comment]
@SimonCropp regarding `BusConfiguration`, that's the issue with running out of sync. ASB dev is on core 6.0.0-unstable1423, which doesn't have `EndpointConfig`.

File: samples/azure/polymorphic-events-asb/Version_6/Shared/ConfigErrorQueue.cs (hunk 3 above)
Comment (SeanFeldman) [inline samples/azure/polymorphic-events-asb/Version_6/Shared/ConfigErrorQueue.cs]: This _is_ code config ;) Looks like we've got issue with naming things.
Perhaps you mean to use 
```c#
busConfiguration.SendFailedMessagesTo("error");
busConfiguration.AuditProcessedMessagesTo("audit");
```

**Comment by SeanFeldman** [comment]
Applied all changes & squashed. @SimonCropp please have a look. Thanks.

File: samples/azure/polymorphic-events-asb/Version_6/Publisher/app.config [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
```
Comment (SimonCropp) [inline samples/azure/polymorphic-events-asb/Version_6/Publisher/app.config]: Delete this file

File: samples/azure/polymorphic-events-asb/Version_6/Shared/app.config [hunk 6]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
```
Comment (SimonCropp) [inline samples/azure/polymorphic-events-asb/Version_6/Shared/app.config]: Delete this file

File: samples/azure/polymorphic-events-asb/Version_6/Subscriber/app.config [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  </configSections>
+
+  <UnicastBusConfig>
+    <MessageEndpointMappings>
+      <add Assembly="Shared" Endpoint="Samples.ASB.Polymorphic.Publisher" />
+    </MessageEndpointMappings>
+  </UnicastBusConfig>
+
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
```
Comment (SimonCropp) [inline samples/azure/polymorphic-events-asb/Version_6/Subscriber/app.config]: Delete this line

**Pull request merged**